src/SaveNicknameToDatabase.php

```diff
@@ -2,23 +2,34 @@
 
 namespace Flarum\Nicknames;
 
+use Flarum\Settings\SettingsRepositoryInterface;
 use Flarum\User\Event\Saving;
+use Flarum\User\Exception\PermissionDeniedException;
 use Illuminate\Support\Arr;
 
 class SaveNicknameToDatabase {
+    /**
+     * @var SettingsRepositoryInterface
+     */
+    protected $settings;
+
+    public function __construct(SettingsRepositoryInterface $settings) {
+        $this->settings = $settings;
+    }
+
     public function handle(Saving $event)
     {
         $user = $event->user;
         $data = $event->data;
         $actor = $event->actor;
 
-        $isSelf = $actor->id === $user->id;
+        $isSelf = $actor->id == $user->id;
         $attributes = Arr::get($data, 'attributes', []);
 
         if (isset($attributes['nickname'])) {
             if ($isSelf) {
                 $actor->assertCan('editOwnNickname', $user);
-            } else {
+            } elseif ($actor->exists) {
                 $actor->assertCan('edit', $user);
```

Comment:
also, doesn't then`edit` ability not exist anymore ? we split it into (credentials, username, groups)

Reply:
We still have `user.edit` for BC purposes, as well as all the other stuff extensions add that doesn't fall under those categories (like nickname!)